guides/Gradio_ONNX_on_Hugging_Face.md

```diff
@@ -0,0 +1,142 @@
+
+### Hugging Face Spaces
+
+Hugging Face Spaces is a free hosting option for Gradio demos. Spaces comes with 3 SDK options: Gradio, Streamlit and Static HTML demos. Spaces can be public or private and the workflow is similar to github repos. There are over 2000+ spaces currently on Hugging Face. To learn more about spaces see: https://huggingface.co/spaces/launch
+
+### Hugging Face Models
+
+Hugging Face Model Hub also supports ONNX models and ONNX models can be filtered through the [ONNX tag](https://huggingface.co/models?library=onnx&sort=downloads)
+
+## How did Hugging Face help ONNX Model Zoo?
+There are a lot of Jupyter notebooks in ONNX Model Zoo for users to test ONNX Model Zoo models. Previously users needed to download the models themselves and run those notebooks locally for testing. With Hugging Face, the testing process can be much simpler and more user-friendly. Users can easily try certain ONNX Model Zoo model on Hugging Face Spaces and run a quick demo powered by Gradio with ONNX Runtime, all on cloud without downloading anything locally. Note that there are various runtimes for ONNX, e.g., [ONNX Runtime](https://github.com/microsoft/onnxruntime), [MXNet](https://github.com/apache/incubator-mxnet).
+
+## What is role of ONNX Runtime?
+ONNX Runtime is a cross-platform inference and training machine-learning accelerator. It makes live Gradio demo with ONNX Model Zoo model on Hugging Face possible.
+
+ONNX Runtime inference can enable faster customer experiences and lower costs, supporting models from deep learning frameworks such as PyTorch and TensorFlow/Keras as well as classical machine learning libraries such as scikit-learn, LightGBM, XGBoost, etc. ONNX Runtime is compatible with different hardware, drivers, and operating systems, and provides optimal performance by leveraging hardware accelerators where applicable alongside graph optimizations and transforms. For more information please see the [official website](https://onnxruntime.ai/).
+
+## Setting up a Gradio Demo for EfficientNet-Lite4
+
+EfficientNet-Lite 4 is the largest variant and most accurate of the set of EfficientNet-Lite model. It is an integer-only quantized model that produces the highest accuracy of all of the EfficientNet models. It achieves 80.4% ImageNet top-1 accuracy, while still running in real-time (e.g. 30ms/image) on a Pixel 4 CPU. To learn more read the [model card](https://github.com/onnx/models/tree/main/vision/classification/efficientnet-lite4)
+
+Here we walk through setting up a example demo for EfficientNet-Lite4 using Gradio
+
+First we import our dependencies and download and load the efficientnet-lite4 model from the onnx model zoo. Then load the labels from the labels_map.txt file. We then setup our preprocessing functions, load the model for inference, and setup the inference function. Finally, the inference function is wrapped into a gradio inferface for a user to interact with. See the full code below.
```

Comment:
Instead of one giant block with the full code i would recommend splitting into the different steps. 

show code for loading the labels, setting up preprocessing, loading the model ...etc

Reply:
looking at past guides, the code to run the entire gradio app.py is in one code block so opted for that, I think we should keep that so users can copy paste the whole block to run a gradio demo if they want to